docs/concepts/native-interop.rst

```diff
@@ -0,0 +1,254 @@
+Native interoperability
+=======================
+
+In this document we will dive a little bit deeper into all three ways of doing 
+"native interoperability" that are available on the .NET platform. 
+
+There are a couple of reasons why you would want to call into native code:
+
+* Operating Systems come with a large volume of APIs that are 
+  not present in the managed class libraries. A prime example for this would be 
+  access to hardware or operating system management functions.
+* Communicating with components that are written in unmanaged languages usually 
```

Comment:
It's a bit odd to say "something is for unmanaged languages ... and this covers managed languages".

It's best to avoid the term "unmanaged", but use the term native. It's really languages that support the C ABI.

Java typically requires JNI to interop via C ABI. I don't think P/Invoke on its own enables interop with Java. Is there an example of that?